### PR TITLE
Just some misc. refactoring of stuff

### PR DESCRIPTION
**Pull request by adigitalmonk**
I put together some misc. refactorings.  If you don't like any of it, let me know and I can remove it or you can just steal stuff into other branches.

### CODE REVIEW

File: server/models/players.js [hunk 1]
```diff
@@ -1,23 +1,19 @@
...
-let Schema = mongoose.Schema;
+const factory = require('./database.js');
 
-let PlayersSchema = new Schema({
+const schema = {
     name: String,
     nickname: String,
-    chant: String,
-    wins: Number,
-    losses: Number
```
Comment (jaydeluca) [inline server/models/players.js]: wins and losses didn't make the cut? was that intentional?

Comment (adigitalmonk) [reply server/models/players.js]: Oh interesting, you know why that probably happened?  I started this, then got up to date after.  Good catch!

**Comment by adigitalmonk** [comment]
Never hesitate! I actually have the environment set up now, so I did test it.  I'm not 100% on all of the functionality, and I did a lot of this as more of a nice-to-have so feel free to push changes or stuff you don't want.

**Comment by ianarsenault** [comment]
@adigitalmonk Does editing a player work for you, because currently when I edit a player it removes all key - values except for id

**Comment by adigitalmonk** [comment]
That just sounds awful.  In master or this branch?

**Comment by ianarsenault** [comment]
Looks like both branches for me.

**Comment by ianarsenault** [comment]
So if you update a player it subsequently breaks the autocomplete 

Example, this is after an edit of a player 

Image

File: server/routes/players.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    app.get('/player/:id', (req, res) => {
+        Players.fetchOne(req.params.id).then(
+            (player) => { res.send(player); },
+            (err) => { console.error(err); } 
+        );
+    });
+
+    // Update a player
+    app.put('/players/:id', (req, res) => {
+        Players.updateOne(req.params.id, req.body.name, req.body.nickname, req.body.chant).then(
```
Comment (adigitalmonk) [inline server/routes/players.js]: This got merged but this was broken still.  lol

Comment (ianarsenault) [reply server/routes/players.js]: Yeah, I realized that immediately afterwards lol.

Comment (adigitalmonk) [reply server/routes/players.js]: I'm making a new branch, no worries.  I should have fixed it in both at the samt ime.

Comment (ianarsenault) [reply server/routes/players.js]: I'll fix it

**Pull request merged**